src/main/java/com/google/cloud/tools/gradle/appengine/core/CloudSdkDownloader.java

```diff
@@ -17,11 +17,199 @@
+      throw new GradleException(e.getMessage());
+    }
+
+    ProgressListener progressListener =
+        new ProgressListener() {
+          @Override
+          public void start(String message, long totalWork) {
+            if (logger != null) {
+              logger.lifecycle(message);
+            }
+          }
+
+          @Override
+          public void update(long workDone) {
+            // TODO: Show progress
+          }
+
+          @Override
+          public void update(String message) {
+            if (logger != null) {
+              logger.lifecycle(message);
+            }
+          }
+
+          @Override
+          public void done() {}
+
+          @Override
+          public ProgressListener newChild(long allocation) {
+            return new ProgressListener() {
+              @Override
+              public void start(String message, long totalWork) {}
+
+              @Override
+              public void update(long workDone) {}
+
+              @Override
+              public void update(String message) {}
+
+              @Override
+              public void done() {}
+
+              @Override
+              public ProgressListener newChild(long allocation) {
+                return null;
+              }
+            };
+          }
+        };
+
+    ConsoleListener consoleListener =
+        new ConsoleListener() {
+          @Override
+          public void console(String rawString) {}
+        };
+
+    try {
+      // Install sdk if not installed
+      if (!sdk.isInstalled()) {
+        if (installer == null) {
+          // Use managed sdk to create SdkInstaller if one isn't provided
+          installer = sdk.newInstaller();
+        }
+        installer.install(progressListener, consoleListener);
+      }
+
+      // Install app engine component
+      if (!sdk.hasComponent(SdkComponent.APP_ENGINE_JAVA)) {
+        if (componentInstaller == null) {
+          componentInstaller = sdk.newComponentInstaller();
+        }
+        componentInstaller.installComponent(
+            SdkComponent.APP_ENGINE_JAVA, progressListener, consoleListener);
+      }
+
+      // If version is set to LATEST, update Cloud SDK
+      if (version.equals(Version.LATEST.getVersion()) && !sdk.isUpToDate()) {
+        if (updater == null) {
+          updater = sdk.newUpdater();
+        }
+        updater.update(progressListener, consoleListener);
+      }
+    } catch (IOException
+        | SdkInstallerException
+        | ManagedSdkVersionMismatchException
+        | InterruptedException
+        | CommandExecutionException
+        | CommandExitException
+        | ManagedSdkVerificationException e) {
+      throw new GradleException(e.getMessage());
+    }
+
+    return sdk.getSdkHome().toFile();
+  }
+
+  /**
+   * Validates the Cloud SDK.
+   *
+   * @return true if the sdk in specified home directory is valid.
+   */
+  public boolean isSdkValid(String version, File home) {
```

Comment:
If this class is only validating SDKs that are user specified, it does not appear like we should include it as part of the sdk downloader class. Validation might even need to happen in another task (checkCloudSdk)?